cmake/compile_flags.cmake

```diff
@@ -127,6 +127,12 @@ FUNCTION(ADD_LINUX_DEB_FLAGS)
       MESSAGE(FATAL_ERROR
         "WITH_PACKAGE_FLAGS=on but dpkg-buildflags --get failed")
     ENDIF()
+
+    # remove "-flto=auto -ffat-lto-objects" returned on Ubuntu 22.04
```

Comment:
I know that we are not using it but upstream has WITH_LTO cmake option. Does this change interfere with it?

Reply:
This fix only removes LTO flags from `dpkg-buildflags` so should not interfer with `WITH_LTO`. Ubuntu 22.04 Jammy returns:
```
$ dpkg-buildflags --get CXXFLAGS
-g -O2 -ffile-prefix-map=/data/mysql-server=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security
```

Ubuntu 20.04 returns:
```
$ dpkg-buildflags --get CXXFLAGS
-g -O2 -fdebug-prefix-map=/data/docker=. -fstack-protector-strong -Wformat -Werror=format-security
```